redefinitions.lua

```diff
@@ -5,64 +5,78 @@ Copyright © 2011-2020 Hugo Locurcio and contributors.
-		{'default:steel_ingot', 'default:steel_ingot', 'default:steel_ingot'},
-		{'', 'group:stick', ''},
-	}
-})
+-- Redefine some of the default crafting recipes to be more productive
 
-minetest.clear_craft({
-	recipe = {
-		{'group:wood', 'group:wood', 'group:wood'},
-		{'group:wood', 'group:wood', 'group:wood'},
-		{'', 'group:stick', ''},
+-- Auxiliary function: take a recipe as returned by get_all_craft_recipes
+-- and turn it into a table that can be used to clear a craft or declare a new one
+local reconstruct_internal_craft = function(recipe)
+	local recp = {
+		{ "", "", "" },
+		{ "", "", "" },
+		{ "", "", "" },
 	}
-})
+	for idx, item in pairs(recipe.items) do
+		local row = math.ceil(idx / recipe.width)
+		local col = idx - (row-1)*recipe.width
+		recp[row][col] = item
+	end
+	return recp
+end
 
-minetest.register_craft({
-	output = 'default:sign_wall_steel 4',
-	recipe = {
-		{'default:steel_ingot', 'default:steel_ingot', 'default:steel_ingot'},
-		{'default:steel_ingot', 'default:steel_ingot', 'default:steel_ingot'},
-		{'', 'group:stick', ''},
-	}
-})
+-- Change the amount produced by recipe by apply func to the old amount
+local change_recipe_amount = function(product, recipe, func)
+	local recp = reconstruct_internal_craft(recipe)
 
-minetest.register_craft({
-	output = 'default:sign_wall_wood 4',
-	recipe = {
-		{'group:wood', 'group:wood', 'group:wood'},
-		{'group:wood', 'group:wood', 'group:wood'},
-		{'', 'group:stick', ''},
-	}
-})
+	local oldamount = tonumber(recipe.output:match(" [0-9]+$") or "1")
 
+	local newamount = func(oldamount)
 
-minetest.clear_craft({
-	recipe = {
-		{"default:papyrus", "default:papyrus", "default:papyrus"}
-	}
-})
-minetest.register_craft({
-	output = "default:paper 4",
-	recipe = {
-		{"default:papyrus", "default:papyrus", "default:papyrus"},
-	}
-})
+	-- remove old crafting recipe
+	local redo = { recipe = recp }
+	minetest.clear_craft(redo)
 
+	-- new output
+	redo.output = ("%s %d"):format(product, newamount)
+	minetest.register_craft(redo)
 
-minetest.register_craft({
-	output = "default:rail 24",
-	recipe = {
-		{"default:steel_ingot", "", "default:steel_ingot"},
-		{"default:steel_ingot", "default:stick", "default:steel_ingot"},
-		{"default:steel_ingot", "", "default:steel_ingot"},
-	}
+	minetest.log("action", ("[MOD]%s: recipe for %s production: %d => %d"):format(modname, product, oldamount, newamount))
+end
+
+local increase_craft_production = function(product, func)
+	local recipes = minetest.get_all_craft_recipes(product)
+	for _, r in pairs(recipes) do
+		if r.type == "normal" or r.method == "normal" then
+			change_recipe_amount(product, r, func)
+		end
+	end
+end
+
+-- Increase the crafting production according to the rules from the table, which is in the form:
+-- {
+--   { detector, amount changing function }
+--   { detector, amount changing function }
+-- }
+-- TODO consider exporting this function to other mods
```

Comment:
```suggestion
-- TODO: consider exporting this function to other mods
```

Reply:
While we're at it, I could just do the export. Does moreblocks provide some kind of API object? Or should I just hook it into minetest directly?

Reply:
@Oblomov Not that I know of.

Reply:
I guess we'll keep it as a TODO for now. Force-pushed with the colon.

Reply:
@MazepaDeni